### PR TITLE
Add section on containers

### PR DESCRIPTION
**Pull request by CHrlS98**
Here is my first pass. Ready for review!

### CODE REVIEW

File: source/arriving/setup_computer.rst [hunk 1]
```diff
@@ -152,12 +152,54 @@ In any case, please refer to the `Github page <https://github.com/scilus/scilpy>
 Containers
 """"""""""
 
-TODO
+Container technologies allow applications and services to be packaged into isolated, self-contained units that can be deployed on any cloud or local host. They are like tiny isolated computers running on other computers. For more information, visit the guide on containers and virtual machines (:ref:`ref_virtualmachines`). A container bundling all the dependencies required for running any flow (see section :ref:`ref_flow`) developped in the lab is available `here <https://hub.docker.com/r/scilus/scilus>`__. In the lab, we use Docker and Apptainer (formerly Singularity) containers, which we describe below.
+
+Docker
+------
+The most popular container technology is undoubtedly Docker. Docker is supported on Mac, Windows and Linux, although it'd seem like Docker running a Linux instance do not work well on MacOS with M1/M2 CPU. The documentation to install Docker is easy enough to follow `(here) <https://docs.docker.com/engine/install/ubuntu>`__.
```
Comment (karanphil) [inline source/arriving/setup_computer.rst:159]: I prefer not to use the "(here)" format for links. Maybe something like "Follow this `documentation <https://docs.docker.com/engine/install/ubuntu>`__ to install Docker".

File: source/arriving/setup_computer.rst [hunk 2]
```diff
@@ -152,12 +152,54 @@ In any case, please refer to the `Github page <https://github.com/scilus/scilpy>
...
+
+.. code-block:: bash
+
+    sudo apt update
+    sudo apt install -y software-properties-common
+    sudo add-apt-repository -y ppa:apptainer/ppa
+    sudo apt update
+    sudo apt install -y apptainer
+
+Additional information can be found on the official docummentation `(here) <https://apptainer.org/docs/admin/main/installation.html#install-ubuntu-packages>`__.
```
Comment (karanphil) [inline source/arriving/setup_computer.rst:194]: Same here, I'd prefer something like "on the official `documentation <https://apptainer.org/docs/admin/main/installation.html#install-ubuntu-packages>`__.".

File: source/arriving/setup_computer.rst [hunk 3]
```diff
@@ -152,12 +152,54 @@ In any case, please refer to the `Github page <https://github.com/scilus/scilpy>
...
+Launch :bash:`docker --version` to verify if it is installed correctly.
+
+Apptainer
+---------
+
+Apptainer (formerly Singularity) is an alternative to Docker. Docker requires root privileges and, as such, is not available on High Performance Computers from the Digital Alliance of Canada (see section :ref:`ref_highperfcomputer`). However, Apptainer containers can run without elevated privileges or root access. Apptainer is particularly useful for saving a Docker container to a file which is then useable in High performance computers, without root privileges.
+
+*Do not try to install on MacOS*
+
+The steps for installing Apptainer on Ubuntu are given below.
```
Comment (karanphil) [inline source/arriving/setup_computer.rst:184]: Maybe add "Please refer to the :ref:`ref_highperfcomputer` section if you intend to use Apptainer on High Performance Computers from the Digital Alliance of Canada."

**Review by karanphil — COMMENTED** [review]
Good job! I have only three small comments. Also, I think you should remove the "Installations" sections from the Virtual Machines page (except for the Virtual Machine section).

File: source/arriving/setup_computer.rst (hunk 3 above)
Comment (CHrlS98) [reply source/arriving/setup_computer.rst:184]: Looks to me like we don't say much on containers in the high performance computer section. HPC are already referred on line 180. Sure you want to add this?

Comment (karanphil) [reply source/arriving/setup_computer.rst:184]: Maybe my sentence is not the right one, but we don't want people to try and install apptainer on Beluga. I agree that it is not clear in the HPC section, but there is a line for module load apptainer. Maybe just say "Do not try these lines on..."

Comment (CHrlS98) [reply source/arriving/setup_computer.rst]: Ok I get it. See my latest commit.

File: source/arriving/setup_computer.rst [hunk 4]
```diff
@@ -152,12 +152,52 @@ In any case, please refer to the `Github page <https://github.com/scilus/scilpy>
 Containers
 """"""""""
 
-TODO
+Container technologies allow applications and services to be packaged into isolated, self-contained units that can be deployed on any cloud or local host. They are like tiny isolated computers running on other computers. For more information, visit the guide on containers and virtual machines (:ref:`ref_virtualmachines`). A container bundling all the dependencies required for running any flow (see section :ref:`ref_flow`) developped in the lab is available `here <https://hub.docker.com/r/scilus/scilus>`__. In the lab, we use Docker and Apptainer (formerly Singularity) containers, which we describe below.
+
+Docker
+------
+The most popular container technology is undoubtedly Docker. Docker is supported on Mac, Windows and Linux, although it'd seem like Docker running a Linux instance do not work well on MacOS with M1/M2 CPU. To install Docker, follow this documentation: `<https://docs.docker.com/engine/install/ubuntu>`__.
```
Comment (karanphil) [inline source/arriving/setup_computer.rst:159]: You prefer fully visible links rather than a clickable "documentation"?

Comment (CHrlS98) [reply source/arriving/setup_computer.rst:159]: No I thought it was what you preferred. Will fix.

Comment (karanphil) [reply source/arriving/setup_computer.rst:159]: I just didn't like the ": here"!

**Review by karanphil — APPROVED** [review]
GTG!

**Pull request merged**